packages/adp-tooling/src/types.ts

```diff
@@ -138,3 +142,149 @@ export const enum HttpStatusCodes {
     NOT_IMPLEMETED = 501,
     SERVICE_UNAVAILABLE = 503
 }
+
+export type Writer = new (fs: Editor, projectPath: string) => IWriter<any>;
+export type IWriterData<T extends GeneratorType> = IWriter<GeneratorData<T>>;
+export interface IWriter<T extends BaseData> {
+    write(data: T): Promise<void>;
+}
+
+export const enum GeneratorType {
+    ADD_ANNOTATIONS_TO_ODATA = 'Add Annotations to OData',
+    ADD_COMPONENT_USAGES = 'Add Component Usages',
+    ADD_NEW_MODEL = 'Add New Model',
+    CHANGE_DATA_SOURCE = 'Change Data Source',
+    CHANGE_INBOUND = 'Change Inbound'
+}
+
+export const enum GeneratorName {
```

Comment:
What is the need for the specific generator names? In the end they are only used when writing the change, correct? If yes, could we simply write `@sap-ux/adp-tooling` in all of them?

Reply:
I think that when you pass incorrect data in a particular manifest change or the change is in the incorrect format, then the support.generator will help us understand which generator has produced the wrong change. 

Also, there might be issues if you write @sap-ux/adp-tooling, maybe the generator name is taken into account server-side when calling the API endpoint to get the merged manifest.

Reply:
Changed in: [3ed676b](https://github.com/SAP/open-ux-tools/pull/1643/commits/3ed676b560a71c69eac49e8258cf84f78c948ade)